specification/hybridconnectivity/resource-manager/Microsoft.HybridConnectivity/PublicCloud/stable/2024-12-01/publicCloud.json

```diff
@@ -0,0 +1,1671 @@
+      "type": "object",
+      "description": "Definition of Solution type resource.",
+      "properties": {
+        "solutionType": {
+          "type": "string",
+          "description": "The name of the solution type."
+        },
+        "description": {
+          "type": "string",
+          "description": "Short description of solution type."
+        },
+        "supportedAzureRegions": {
+          "type": "array",
+          "description": "The locations this solution is supported in.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "solutionSettings": {
+          "type": "array",
+          "description": "Array of solution settings and its description.",
+          "items": {
+            "$ref": "#/definitions/SolutionTypeSettingsProperties"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "SolutionTypeResource": {
+      "type": "object",
+      "description": "Concrete proxy resource types can be created by aliasing this type using a specific property type.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SolutionTypeProperties",
+          "description": "The resource-specific properties for this resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "SolutionTypeResourceListResult": {
+      "type": "object",
+      "description": "The response of a SolutionTypeResource list operation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The SolutionTypeResource items on this page",
+          "items": {
+            "$ref": "#/definitions/SolutionTypeResource"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "SolutionTypeSettings": {
+      "type": "object",
+      "description": "The properties of Solution Type",
+      "properties": {
+        "solutionType": {
+          "type": "string",
+          "description": "The type of the solution"
+        },
+        "solutionSettings": {
+          "$ref": "#/definitions/SolutionSettings",
+          "description": "Solution settings"
+        }
+      },
+      "required": [
+        "solutionType"
+      ]
+    },
+    "SolutionTypeSettingsProperties": {
+      "type": "object",
+      "description": "Represent Solution settings properties description array.",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the solution setting property."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "The UI friendly name of the solution setting property."
+        },
+        "type": {
```

Comment:
>type [](http://example.com/codeflow?start=9&length=4)

can this be from a set of enums ?

Reply:
here the type refers to data type of the settings, not sure if we want this as enum